src/ai.cpp

```diff
@@ -8,15 +8,34 @@ bool AI::init()
 
 void AI::make_villains_decide(std::map<direction, Lane*> lanes)
 {
+	std::map<direction, Lane*> new_lanes = deep_copy_lanes(lanes);
+	std::map<direction, int> cars_in_intersection;
+
+	// Prune cars in the intersection
```

Comment:
This fixes bug (1). We need to remove all cars that are in the intersection as they are no longer important from an AI point of view. Before doing this, the car at the front of the line could be considered the second car in the lane and thus eligible for villain decision making.